packages/introspection/src/minifyIntrospectionQuery.ts

```diff
@@ -0,0 +1,153 @@
+import {
+  IntrospectionQuery,
+  IntrospectionType,
+  IntrospectionTypeRef,
+} from 'graphql';
+
+const anyType: IntrospectionTypeRef = {
+  kind: 'SCALAR',
+  name: 'Any',
+};
+
+const mapType = (
+  fromType: any,
+  toType: IntrospectionTypeRef
+): IntrospectionTypeRef => {
+  switch (fromType.kind) {
+    case 'NON_NULL':
+    case 'LIST':
+      return {
+        kind: fromType.kind,
+        ofType: mapType(fromType.ofType, toType),
+      };
+
+    case 'SCALAR':
+    case 'INPUT_OBJECT':
+    case 'ENUM':
+      return toType;
+
+    case 'OBJECT':
+    case 'INTERFACE':
+    case 'UNION':
+      return {
+        kind: fromType.kind,
+        name: fromType.name,
+      };
+
+    default:
+      throw new TypeError(
+        `Unrecognized type reference of type: ${(fromType as any).kind}.`
+      );
+  }
+};
+
+const minifyIntrospectionType = (
+  type: IntrospectionType
+): IntrospectionType => {
+  switch (type.kind) {
+    case 'OBJECT': {
```

Comment:
Can we maybe use the `Kind` exported from graphql for safety?

Reply:
I'm not sure if it's the same 🤔

Reply:
Oh all right! thanks for checking 👍